sips/sip-013/sip-013-semi-fungible-token-standard.md

```diff
@@ -0,0 +1,248 @@
+
+```JSON
+{
+	"sip": 16
+	// ... any other properties
+}
+```
+
+Applications consuming these metadata files can base display capabilities on the `sip` value. It should refer to a SIP number describing a JSON metadata standard.
+
+### Transfer
+
+`(transfer ((token-id uint) (amount uint) (sender principal) (recipient principal)) (response bool uint))`
+
+Transfer a token from the sender to the recipient. It is recommended to leverage Clarity primitives like `ft-transfer?` to help safeguard users. The function should return `(ok true)` on success or an `err` response containing an unsigned integer on failure. The failure codes follow the existing conventions of `stx-transfer?` and `ft-transfer?`.
+
+| Error code | Description                                      |
+|------------|--------------------------------------------------|
+| `u1`       | The sender has insufficient balance.             |
+| `u2`       | The sender and recipient are the same principal. |
+| `u3`       | Amount is `u0`.                                  |
+| `u4`       | The sender is not authorised to transfer tokens. |
+
+Error code `u4` is broad and may be returned under different cirumstances. For example, a token  contract with an allowance mechanism can return `(err u4)` when the `sender` parameter has no allowance for the specified token amount or if the sender is not equal to `tx-sender` or `contract-owner`. A token contract without an allowance mechanism can return `(err u4)` simply when the `sender` is not what is expected.
+
+Since it is possible for smart contracts to own tokens, it is recommended to check for both `tx-sender` and `contract-caller` as it allows smart contracts to transfer tokens it owns without having to resort to using `as-contract`. Such a guard can be constructed as follows:
+
+```clarity
+(asserts! (or (is-eq sender tx-sender) (is-eq sender contract-caller)) (err u4))
+```
+
+The `transfer` function should emit a special transfer event, as detailed in the Events section of this document.
+
+### Transfer with memo
+
+`(transfer-memo ((token-id uint) (amount uint) (sender principal) (recipient principal) (memo (buff 34))) (response bool uint))`
+
+Transfer a token from the sender to the recipient and emit a memo. This function follows the exact same procedure as `transfer` but emits the provided memo via `(print memo)`. The memo event should be the final event emitted by the contract. (See also the events section of this document below.)
+
+### Bulk transfers
+
+`(transfer-many ((transfers (list 100 {token-id: uint, amount: uint, sender: principal, recipient: principal}))) (response bool uint))`
```

Comment:
If there's going to be a cap of 100 transfers here, why not just call this `transfer-100`?

Reply:
Same for `transfer-many-memo` below.

Reply:
I was mirroring the naming convention of the `send-many` STX tool. I can see how adding the total can be useful so that a `transfer-N` can be introduced by a future SIP. Having said that, I want to see if there are any contracts on mainnet that already implement this draft SIP in its current form.

Additionally, I decided to change it back to its original value of `200`. While it might not be possible to call it with a full list right now, if we anticipate this to become a standard and used for many years to come, one can assume that we can see another runtime cost adjustment at some point in the future.

Reply:
Yeah, you could make this number arbitrarily large for forward-compatibility.  The block limit probably won't increase by more than a factor of 100, but you never know.